examples/stocks/test/icon_color_test.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:ui' as ui show window;
+
+import 'package:test/test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:stocks/main.dart' as stocks;
+import 'package:stocks/stock_data.dart' as stock_data;
+
+Element findElementOfExactWidgetTypeGoingDown(Element node, Type targetType) {
+  void walker(Element child) {
+    if (child.widget.runtimeType == targetType)
+      throw child;
+    child.visitChildElements(walker);
+  }
+  try {
+    walker(node);
+  } on Element catch (result) {
+    return result;
+  }
+  return null;
+}
+
+Element findElementOfExactWidgetTypeGoingUp(Element node, Type targetType) {
+  Element result;
+  bool walker(Element ancestor) {
+    if (ancestor.widget.runtimeType == targetType)
+      result = ancestor;
+    return result == null;
+  }
+  node.visitAncestorElements(walker);
+  return result;
+}
+
+final RegExp materialIconAssetNameColorExtractor = new RegExp(r'[^/]+/ic_.+_(white|black)_[0-9]+dp\.png');
+
+void checkIconColor(WidgetTester tester, String label, Color color) {
+  // The icon is going to be in the same merged semantics box as the text
+  // regardless of how the menu item is represented, so this is a good
+  // way to find the menu item. I hope.
+  Element semantics = findElementOfExactWidgetTypeGoingUp(tester.findText(label), MergeSemantics);
+  expect(semantics, isNotNull);
+  Element asset = findElementOfExactWidgetTypeGoingDown(semantics, AssetImage);
+  RenderImage imageBox = asset.findRenderObject();
+  Match match = materialIconAssetNameColorExtractor.firstMatch(asset.widget.name);
+  expect(match, isNotNull);
+  if (color == const Color(0xFFFFFFFF)) {
+    expect(match[1], equals('white'));
```

Comment:
White icons are common (more so than black ones, it seems). Colourising every icon seems not great for perf. I dunno though. It's causing other problems too -- e.g. we can't lerp between icon colours easily since they're sometimes colourised and sometimes not.